js/app/models/ImportFileWrapperModel.js

```diff
@@ -120,8 +120,8 @@ app.factory('ImportFileWrapper', function(Hook, SplitterService) {
 				context.progressToReach = context.splittedICal.vevents.length +
 					context.splittedICal.vjournals.length +
 					context.splittedICal.vtodos.length;
-				afterReadCallback();
 				iface.state = ImportFileWrapper.stateAnalyzed;
+				afterReadCallback();
```

Comment:
Sure, but it would be nice for the imported calendar to show up directly without having to refresh the page.